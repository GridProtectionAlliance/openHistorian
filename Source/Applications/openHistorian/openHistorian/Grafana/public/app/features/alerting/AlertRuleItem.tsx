import React, { PureComponent } from 'react';
// @ts-ignore
import Highlighter from 'react-highlight-words';
import { AlertRule } from '../../types';
import { Icon, IconName, Button, Tooltip, LinkButton, HorizontalGroup } from '@grafana/ui';

export interface Props {
  rule: AlertRule;
  search: string;
  onTogglePause: () => void;
}

class AlertRuleItem extends PureComponent<Props> {
  renderText(text: string) {
    return (
      <Highlighter
        highlightClassName="highlight-search-match"
        textToHighlight={text}
        searchWords={[this.props.search]}
      />
    );
  }

  render() {
    const { rule, onTogglePause } = this.props;

    const ruleUrl = `${rule.url}?editPanel=${rule.panelId}&tab=alert`;

    return (
      <li className="alert-rule-item">
        <Icon size="xl" name={rule.stateIcon as IconName} className={`alert-rule-item__icon ${rule.stateClass}`} />
        <div className="alert-rule-item__body">
          <div className="alert-rule-item__header">
            <div className="alert-rule-item__name">
              <a href={ruleUrl}>{this.renderText(rule.name)}</a>
            </div>
            <div className="alert-rule-item__text">
              <span className={`${rule.stateClass}`}>{this.renderText(rule.stateText)}</span>
              <span className="alert-rule-item__time"> for {rule.stateAge}</span>
            </div>
          </div>
          {rule.info && <div className="small muted alert-rule-item__info">{this.renderText(rule.info)}</div>}
        </div>

        <div className="alert-rule-item__actions">
          <HorizontalGroup spacing="sm">
            <Tooltip placement="bottom" content="Pausing an alert rule prevents it from executing">
              <Button
                variant="secondary"
                size="sm"
                icon={rule.state === 'paused' ? 'play' : 'pause'}
                onClick={onTogglePause}
              />
            </Tooltip>
            <Tooltip placement="right" content="Edit alert rule">
              <LinkButton size="sm" variant="secondary" href={ruleUrl} icon="cog" />
            </Tooltip>
          </HorizontalGroup>
        </div>
      </li>
    );
  }
}

export default AlertRuleItem;
