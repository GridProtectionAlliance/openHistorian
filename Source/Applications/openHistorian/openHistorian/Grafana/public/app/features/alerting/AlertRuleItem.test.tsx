import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

import AlertRuleItem, { Props } from './AlertRuleItem';

const setup = (propOverrides?: object) => {
  const props: Props = {
    rule: {
      id: 1,
      dashboardId: 1,
      panelId: 1,
      name: 'Some rule',
      state: 'Open',
      stateText: 'state text',
      stateIcon: 'anchor',
      stateClass: 'state class',
      stateAge: 'age',
      url: 'https://something.something.darkside',
    },
    search: '',
    onTogglePause: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return render(<AlertRuleItem {...props} />);
};

describe('AlertRuleItem', () => {
  it('should render component', () => {
    const mockToggle = jest.fn();
    setup({ onTogglePause: mockToggle });

    expect(screen.getByText('Some rule')).toBeInTheDocument();
    expect(screen.getByText('state text')).toBeInTheDocument();
    expect(screen.getByText('Pause')).toBeInTheDocument();
    expect(screen.getByText('Edit alert')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Pause'));
    expect(mockToggle).toHaveBeenCalled();
  });
});
