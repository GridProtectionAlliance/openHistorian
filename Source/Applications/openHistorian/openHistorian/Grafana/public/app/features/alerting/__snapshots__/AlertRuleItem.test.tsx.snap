// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<li
  className="alert-rule-item"
>
  <Icon
    className="alert-rule-item__icon state class"
    name="icon"
    size="xl"
  />
  <div
    className="alert-rule-item__body"
  >
    <div
      className="alert-rule-item__header"
    >
      <div
        className="alert-rule-item__name"
      >
        <a
          href="https://something.something.darkside?editPanel=1&tab=alert"
        >
          <Highlighter
            highlightClassName="highlight-search-match"
            searchWords={
              Array [
                "",
              ]
            }
            textToHighlight="Some rule"
          />
        </a>
      </div>
      <div
        className="alert-rule-item__text"
      >
        <span
          className="state class"
        >
          <Highlighter
            highlightClassName="highlight-search-match"
            searchWords={
              Array [
                "",
              ]
            }
            textToHighlight="state text"
          />
        </span>
        <span
          className="alert-rule-item__time"
        >
           for 
          age
        </span>
      </div>
    </div>
  </div>
  <div
    className="alert-rule-item__actions"
  >
    <Component
      spacing="sm"
    >
      <Component
        content="Pausing an alert rule prevents it from executing"
        placement="bottom"
      >
        <Button
          icon="pause"
          onClick={[MockFunction]}
          size="sm"
          variant="secondary"
        />
      </Component>
      <Component
        content="Edit alert rule"
        placement="right"
      >
        <LinkButton
          href="https://something.something.darkside?editPanel=1&tab=alert"
          icon="cog"
          size="sm"
          variant="secondary"
        />
      </Component>
    </Component>
  </div>
</li>
`;
